chapters/en/chapter7/speech-to-speech.mdx

```diff
@@ -0,0 +1,242 @@
+
+First, let's load the SpeechT5 processor, model and vocoder from the pre-trained checkpoint:
+
+```python
+from transformers import SpeechT5Processor, SpeechT5ForTextToSpeech, SpeechT5HifiGan
+
+processor = SpeechT5Processor.from_pretrained("microsoft/speecht5_tts")
+
+model = SpeechT5ForTextToSpeech.from_pretrained("microsoft/speecht5_tts")
+vocoder = SpeechT5HifiGan.from_pretrained("microsoft/speecht5_hifigan")
+```
+
+<Tip>
+    Here we're using SpeechT5 checkpoint trained specifically for English TTS. Should you wish to translate into a language 
+    other than English, either swap the checkpoint for a SpeechT5 TTS model fine-tuned on your language of choice, or 
+    use an MMS TTS checkpoint pre-trained in your target langauge.
+</Tip>
+
+As with the Whisper model, we'll place the SpeechT5 model and vocoder on our GPU accelerator device if we have one:
+```python
+model.to(device)
+vocoder.to(device)
+```
+
+Great! Let's load up the speaker embeddings:
+
+```python
+embeddings_dataset = load_dataset("Matthijs/cmu-arctic-xvectors", split="validation")
+speaker_embeddings = torch.tensor(embeddings_dataset[7306]["xvector"]).unsqueeze(0)
+```
+
+We can now write a function that takes a text prompt as input, and generates the corresponding speech. We'll first pre-process
+the text input using the SpeechT5 processor, tokenizing the text to get our input ids. We'll then pass the input ids and
+speaker embeddings to the SpeechT5 model, placing each on the accelerator device if available. Finally, we'll return the 
+generated speech, bringing it back to the CPU so that we can play it back in our ipynb notebook:
+
+```python
+def synthesise(text):
+    inputs = processor(text=text, return_tensors="pt")
+    speech = model.generate_speech(
+        inputs["input_ids"].to(device), speaker_embeddings.to(device), vocoder=vocoder
+    )
+    return speech.cpu()
+```
+
+Let's check it works with a dummy text input:
+```python
+speech = synthesise("Hey there! This is a test!")
+
+Audio(speech, rate=16000)
+```
+
+Sounds good! Now for the exciting part - piecing it all together.
+
+## Creating a STST demo
+
+Before we create a [Gradio](https://gradio.app) demo to showcase our STST system, let's first do a quick sanity check 
+to make sure we can concatenate the two models, putting an audio sample in and getting an audio sample out. We'll do 
+this by concatenating the two functions we defined in the previous two sub-sections, such that we input the source audio 
+and retrieve the translated text, then synthesise the translated text to get the translated speech. Finally, we'll convert 
+the synthesised speech to an `int16` array, which is the output audio file format expected by Gradio:
+
+```python
+import numpy as np
+
+
+def speech_to_speech_translation(audio):
+    translated_text = translate(audio)
+    synthesised_speech = synthesise(translated_text)
+    synthesised_speech = (synthesised_speech.numpy() * 32767).astype(np.int16)
```

Comment:
Not sure I understand what the number 32767 here is.

Reply:
Conversion from float32 to int16 - will see if there's a more elegant way to do this rather than hardcoding

Reply:
Resolved in https://github.com/huggingface/audio-transformers-course/pull/84/commits/4238826f4a69ce1ca7d8d5bcfbb3848b7bf715f8